en/docs/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/general-recommendations-for-secure-coding.md

```diff
@@ -1509,6 +2371,106 @@ In summary, when integrating OWASP CSRFGuard with a product, it is required to d
+* Ensure that CSRF tokens are generated using cryptographically secure random number generators.
+* Always use HTTP POST, PUT, DELETE (not GET) for state-changing operations.
+* Implement proper token validation for all state-changing operations.
+
+!!! bug error "Example Unsafe Usage"
+    ```go
+    func createUserHandler(w http.ResponseWriter, r *http.Request) {
+        if r.Method != http.MethodPost {
+            http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+            return
+        }
+        
+        username := r.FormValue("username")
+        password := r.FormValue("password")
+        createUser(username, password)
+        fmt.Fprintf(w, "User created successfully")
+    }
+    ```
+
+!!! success check done "Example Recommended Usage"
+    ```go
+    import (
+        "github.com/gorilla/csrf"
+        "net/http"
+    )
+    
+    func main() {
+        CSRF := csrf.Protect(
+            []byte("32-byte-long-auth-key"),
+            csrf.Secure(true),
+            csrf.HttpOnly(true),
+        )
+        
+        http.HandleFunc("/create_user", createUserHandler)
+        http.ListenAndServe(":8000", CSRF(http.DefaultServeMux))
+    }
+    
+    func createUserHandler(w http.ResponseWriter, r *http.Request) {
+        // CSRF validation happens automatically via the middleware
+        if r.Method != http.MethodPost {
+            http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+            return
+        }
+        
+        username := r.FormValue("username")
+        password := r.FormValue("password")
+        
+        // Process the user creation
+        createUser(username, password)
+        
+        fmt.Fprintf(w, "User created successfully")
+    }
+    ```
+
+!!! success check done "Example Recommended Usage"
+    ```go
+    import (
+        "github.com/justinas/nosurf"
+        "net/http"
+        "html/template"
+    )
+    
+    func main() {
+        mux := http.NewServeMux()
+        mux.HandleFunc("/create_user", createUserHandler)
+        
+        // Wrap the servemux with the nosurf middleware
+        csrfHandler := nosurf.New(mux)
+        
+        // Configure cookie options
+        csrfHandler.SetBaseCookie(http.Cookie{
+            Path:     "/",
+            HttpOnly: true,
+            Secure:   true,
+            SameSite: http.SameSiteStrictMode,
+        })
+        
+        http.ListenAndServe(":8000", csrfHandler)
+    }
+    
+    func createUserHandler(w http.ResponseWriter, r *http.Request) {
+        // nosurf middleware automatically validates the token
+        if r.Method != http.MethodPost {
+            http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+            return
+        }
+        
+        username := r.FormValue("username")
+        password := r.FormValue("password")
+        createUser(username, password)
+        fmt.Fprintf(w, "User created successfully")
+    }
+    ```
```

Comment:
```suggestion
```

Reply:
removing since no need to give multiple options unless there is a good reason.

Reply:
Addressed with 0e04e157f23d87082c1b6dbfc19b97a2e4f0ba45